ts/screens/wallet/AddCardScreen.tsx

```diff
@@ -124,14 +134,27 @@ function getCardFromState(state: State): Option<CreditCard> {
 }
 
 class AddCardScreen extends React.Component<Props, State> {
+  private panRef = React.createRef<typeof MaskedInput>();
+  private expirationDateRef = React.createRef<typeof MaskedInput>();
+  private securityCodeRef = React.createRef<typeof MaskedInput>();
+
   constructor(props: Props) {
     super(props);
-    this.state = {
-      pan: none,
-      expirationDate: none,
-      securityCode: none,
-      holder: none
-    };
+    this.state = INITIAL_STATE;
+  }
+
+  public componentDidMount() {
+    // The AppState change is also stored and notified by our redux store but
+    // we are using the event lister directly because we want to clear the
+    // input fileds as soon as possible before going in background (remember
+    // Android does not have a "inactive" intermediate state).
+    // This prevent the input fields to be cleared only when the application
```

Comment:
```suggestion
    // This prevents the input fields to be cleared only when the application
```

Reply:
not sure about the meaning of this

Reply:
If you wait the AppState change from the store sometimes when you reopen the app the input values are still present and after some milliseconds get cleared.

Reply:
much better explanation than the one you wrote above :D